sdk/include/opentelemetry/sdk/trace/span_data.h

```diff
@@ -197,7 +197,10 @@ class SpanData final : public Recordable
   void SetAttribute(nostd::string_view key,
                     const opentelemetry::common::AttributeValue &value) noexcept override
   {
-    attribute_map_.SetAttribute(key, value);
+    if (attribute_map_.GetAttributeMapSize() < GetSpanLimits().AttributeCountLimit
+          || attribute_map_.KeyExists(key)) {
+      attribute_map_.SetAttribute(key, value);
+    }
```

Comment:
Not sure if this is a good idea to drop the attributes silently.

Reply:
Agreed. 
We can either log every time when attribute is dropped, but that would be excessive logging and is not recommended in the spec.
Second way, we can store the information of how many attributes are added in the span and so the number of attributes dropped. This should be enough information, any opinion on this?

Reply:
Yeah, this won't really work... Because I'd see that class being used not necessarily only for Span Attributes. But for Resource Attributes as well. Do resource attributes have the same limit? I mean - can we move this check elsewhere.. Can we keep it a generic converter-helper class to transform attributes from API AttributeValue to OwnedAttributeValue, and just keep it as generic as possible, without binding it to Span specifics.

Reply:
The limit is specific to Span attributes, links, and events. SDK doesn't own/copy any of these properties internally, so it is not possible to impose this limit within sdk. All these properties are propagated to exporters, and it is up to the exporters to impose the limits. The exporters may choose not to transform these properties to `OwnedAttributeValue` and instead transform/serialize to (say) JSON format for performance reasons.

Reply:
@lalitb - I think we should have that trigger (limit enforcement) somewhere when we transform into `OwnedAttributeValue`. I don't have concrete proposals. Maybe we should merge the foundation class for some enforcement, proposed here in this PR. Then find out a good way to bind to that class.